includes/controller/user_settings_controller.php

```diff
@@ -187,12 +188,48 @@ function user_settings() {
         $display_message=strip_request_item('display_message');
       else
         $ok = false;
-      if($ok){
+
+      if ($ok){
         update_display_msg($display_message);
         success("Message Changed");
         redirect(page_link_to('user_settings'));
       }
   }
+  elseif (isset($_REQUEST['update'])) {
+    $ok = true;
+
+    $online_ver = file_get_contents("https://raw.githubusercontent.com/fossasia/engelsystem/master/Version.txt");
+    $current_ver = file_get_contents(" ./Version.txt");
+    if (strcmp($current_ver, $online_ver) != 0) {
+      echo shell_exec("sudo sh ./update.sh");
```

Comment:
Instead of echo you can use exec command
http://php.net/manual/en/function.exec.php